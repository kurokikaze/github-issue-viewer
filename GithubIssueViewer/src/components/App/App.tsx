/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * Generated with the TypeScript template
 * https://github.com/react-native-community/react-native-template-typescript
 *
 * @format
 */

import React from 'react';

import {NavigationContainer} from '@react-navigation/native';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import {useColorScheme} from 'react-native';

import {ThemeContext, themes} from '../ThemeContext/ThemeContext';
import RepoSearchScreen from '../../screens/RepoSearchScreen/RepoSearchScreen';
import IssuesBrowserScreen from '../../screens/IssuesBrowserScreen/IssuesBrowserScreen';
import IssueViewerScreen from '../../screens/IssueViewerScreen/IssueViewerScreen';

import {RootStackParamList} from '../../types';

const Stack = createNativeStackNavigator<RootStackParamList>();

const App = () => {
  const isDarkMode = useColorScheme() === 'dark';

  return (
    <ThemeContext.Provider value={isDarkMode ? themes.dark : themes.light}>
      <NavigationContainer>
        <Stack.Navigator initialRouteName="Home">
          <Stack.Screen
            name="Home"
            component={RepoSearchScreen}
            options={{title: 'Search users'}}
          />
          <Stack.Screen
            name="IssuesBrowser"
            component={IssuesBrowserScreen}
            options={{title: 'Browse Issues'}}
          />
          <Stack.Screen
            name="IssueViewer"
            component={IssueViewerScreen}
            options={{title: 'Issue Viewer'}}
          />
        </Stack.Navigator>
      </NavigationContainer>
    </ThemeContext.Provider>
  );
};

export default App;
