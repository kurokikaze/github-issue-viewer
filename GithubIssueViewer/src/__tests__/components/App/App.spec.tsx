import 'react-native';
import React from 'react';
import App from '../../../components/App/App';

// Note: test renderer must be required after react-native.
import {createRenderer} from 'react-test-renderer/shallow';

describe('App', () => {
  it('renders correctly', () => {
    const renderer = createRenderer();
    const wrapper = renderer.render(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
