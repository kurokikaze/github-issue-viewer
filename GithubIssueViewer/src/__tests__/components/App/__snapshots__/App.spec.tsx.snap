// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<Context.Provider
  value={
    Object {
      "alternativeContainerStyle": Object {
        "backgroundColor": "#ddd",
      },
      "barStyle": "light-content",
      "containerStyle": Object {
        "backgroundColor": "#ccc",
      },
      "textStyle": Object {
        "color": "#333",
      },
    }
  }
>
  <ForwardRef(NavigationContainerInner)>
    <NativeStackNavigator
      initialRouteName="IssuesBrowser"
    >
      <Screen
        component={[Function]}
        name="Settings"
        options={
          Object {
            "title": "Settings",
          }
        }
      />
      <Screen
        component={[Function]}
        name="IssuesBrowser"
        options={
          Object {
            "title": "Browse Issues",
          }
        }
      />
      <Screen
        component={[Function]}
        name="IssueViewer"
        options={
          Object {
            "title": "Issue Viewer",
          }
        }
      />
      <Screen
        component={[Function]}
        name="BookmarksBrowser"
        options={
          Object {
            "title": "Issue Viewer",
          }
        }
      />
    </NativeStackNavigator>
  </ForwardRef(NavigationContainerInner)>
</Context.Provider>
`;
